amundsen_application/static/js/components/common/ResourceList/index.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import Pagination from 'react-js-pagination';
+import ResourceListItem from 'components/common/ResourceListItem';
+import { Resource } from 'interfaces';
+import { ITEMS_PER_PAGE, PAGINATION_PAGE_RANGE } from './constants';
+
+
+export interface ResourceListProps {
+  items: Resource[];
+  source: string;
+
+  // The following props are only used if 'paginate' is set to true
+  paginate?: boolean;
```

Comment:
nit: Another common practice for flexible, re-useable components is introduce a property that takes an object for all of the properties for a particular feature. i.e. we'd have 
`paginate: boolean`
`paginateOptions: { ...all those other props }`. 

Fine to merge without it, and perhaps the next time we end up wanting to leverage for of the 3rd party `Pagination` options, we can make that switch.

Reply:
I do like this pattern, but how do the default props work for the nested options? If you specify the `paginateOptions` with just one or two options, does it overwrite all of the fields?

```
defaultProps = {
  paginate: true,
  paginationOptions: {
    activePage: 0,
    isFullList: true,
    itemsPerPage: 4,
  }
}
```

```
<ResourceList 
  paginate={ true }
  paginateOptions={
    itemsPerPage: 5
  }
```

Reply:
ick, looks like React only supports defaultProps for top level properties. Here's a ticket for that request but it was closed, however some folks have posted workarounds: https://github.com/facebook/react/issues/2568. Wouldn't suggest wasting time on this though.